CRM/Utils/PDF/Document.php

```diff
@@ -95,4 +95,64 @@ public static function toTwip($value, $metric) {
     return \PhpOffice\PhpWord\Shared\Converter::pointToTwip($point);
   }
 
+  /**
+   * @param array $path  docx/odt file path
+   * @param string $type  File type
+   * @param bool $returnContent extract content of docx/odt file as a text, later used for token replacement
+   *
+   * @return string
+   *    Return filepath of created html copy of document OR extracted content of document
+   */
+  public static function docReader($path, $type, $returnContent = FALSE) {
+    if ($returnContent) {
+      return self::doc2Text($path, $type);
+    }
+
+    $newpath = str_replace('.' . $type, '.html', $path);
+    $fileType = ($type == 'docx') ? 'Word2007' : 'ODText';
+    if (!file_exists($newpath)) {
+      $phpWord = \PhpOffice\PhpWord\IOFactory::load($path, $fileType);
+      $phpWord->save($newpath, 'HTML');
+    }
+
+    $relPath = str_replace('/Users/monish/www/civicrm-master/', CRM_Utils_System::baseURL(), $newpath);
```

Comment:
@colemanw  sorry for the hardcording but is there any function that turns absolute url to server-sided url ? (didn't found one :( )